### PR TITLE
Minor cleanup in ARCSequenceOpts

### PR DESCRIPTION
**Pull request by meg-gupta**
This change removes some more dead code in ARCSequenceOpts. And adds more info to the LoopRegion printer

- Removes more reference to Insertion points and NewInsts that are no longer needed in ARCSequenceOpts
- Update bottom up loop traversal to remove some code duplication
- Add more info in LoopRegion printer, this is useful while calling dump() utilities within the debugger

### CODE REVIEW

**Comment by meg-gupta** [comment]
@swift-ci test

**Comment by meg-gupta** [comment]
@swift-ci test OS X platform

**Comment by meg-gupta** [comment]
@swift-ci test Linux platform

**Review by atrick — APPROVED** [review]
LGTM, but I don't understand the "make bottom up traversal simpler" commit.

**Review by gottesmm — COMMENTED** [review]
Some thoughts, comments, etc.

File: lib/SILOptimizer/ARC/RefCountState.cpp [hunk 1]
```diff
@@ -285,9 +285,6 @@ handleRefCountInstMatch(SILInstruction *RefCountInst) {
     return false;
   case LatticeState::Decremented:
   case LatticeState::MightBeUsed:
-    // Unset InsertPt so we remove retain release pairs instead of
-    // performing code motion.
```
[the remark this answers is not in the document]
Comment (meg-gupta) [reply lib/SILOptimizer/ARC/RefCountState.cpp:289]: Yes, there were a lot of stale comments left out about 'InsertPt'. This code was refactored out and a lot of functionality moved to ARCCodeMotion.

File: lib/SILOptimizer/Analysis/LoopRegionAnalysis.cpp [hunk 2]
```diff
@@ -88,6 +89,19 @@ void LoopRegion::print(llvm::raw_ostream &os, bool isShort) const {
 
   os << " ucfh:" << (IsUnknownControlFlowEdgeHead? "true " : "false")
      << " ucft:" << (IsUnknownControlFlowEdgeTail? "true " : "false");
+
+  if (isVerbose) {
```
[the remark this answers is not in the document]
Comment (meg-gupta) [reply lib/SILOptimizer/Analysis/LoopRegionAnalysis.cpp:93]: Yes! Sorry I needed to be reminded :)

File: lib/SILOptimizer/ARC/GlobalLoopARCSequenceDataflow.cpp [hunk 3]
```diff
@@ -200,40 +200,14 @@ bool LoopARCSequenceDataflowEvaluator::processLoopBottomUp(
...
-
-    LLVM_DEBUG(llvm::dbgs() << "Merging Successors!\n");
-    mergeSuccessors(Subregion, SubregionData);
-
-    // Then perform the region optimization.
-    NestingDetected |= SubregionData.processBottomUp(
-        AA, RCFI, EAFI, LRFI, FreezeOwnedArgEpilogueReleases, IncToDecStateMap,
-        RegionStateInfo, SetFactory);
-    --End;
-  }
```
[the remark this answers is not in the document]
Comment (meg-gupta) [reply lib/SILOptimizer/ARC/GlobalLoopARCSequenceDataflow.cpp:226]: I can't tell much from history. Maybe just an oversight that it wasn't updated with getReverseSubregions.

**Comment by meg-gupta** [comment]
@swift-ci smoke test

File: lib/SILOptimizer/ARC/GlobalLoopARCSequenceDataflow.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (gottesmm) [reply lib/SILOptimizer/ARC/GlobalLoopARCSequenceDataflow.cpp:226]: I remember why I think. I think there was a miscompile from clang with getReverseSubregions() when I wrote this. So I worked around it. Thanks for fixing!

**Pull request merged**